portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -60,12 +60,22 @@ private int getOffset(HttpServletRequest request) {
     return getNonNegativeIntParameter(request, "offset", DEFAULT_COMMENT_OFFSET);
   }
 
+  /**
+   * Returns the list of comments translated.
+   */
+  private List<Comment> loadAndTranslateComments(HttpServletRequest request) {
+    List<Comment> comments = CommentsStore.load(getLimit(request), getOffset(request));
+    String languageCode = getParameter(request, "language", "en");
+    comments.forEach((comment) -> comment.translate(languageCode));
+    return comments;
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType(MediaType.APPLICATION_JSON);
     response.getWriter().println(
```

Comment:
Tor readability I recommend putting each separate step onto a separate line and assign it to a variable with a good name.

Reply:
Do you mean separate `response.getWriter()` and `println()` or what is inside `println()`?

Reply:
What's inside println.

Reply:
Maybe something like:
```
List<Comments> comments = loadComments(limit, offset);
List<Comments> translatedComments = translateComments(comments, targetLangauge);
sendJsonResponse(response, comments);
```